### DISCUSSION TITLE
How can you extend thunk MiddlewareApi?

### DISCUSSION THREAD

**Discussion by Eonfuzz** [reduxjs/redux-toolkit#4232] [Q&A] [answered]
Heyo Redux team / community!

Right now I'm looking at implementing a hybrid of offline/online first design for an app, and wondering if there's a nice way I can extend the middlewareApi object to include refs to my DB and offline status.

I want to do something like this:
![image](https://github.com/reduxjs/redux-toolkit/assets/5878385/4e646275-d7f6-46c5-bc41-f7c870bb780c)

If not, how would you go about injecting objects into every thunk, or would you recommend a different approach?

**Comment by EskiMojo14** [comment, marked as answer]
you can provide an extra argument to thunks when initialising your store, which would be thunkApi.extra:
https://redux-toolkit.js.org/api/getDefaultMiddleware#customizing-the-included-middleware

**Reply by Eonfuzz** [reply]
Awesome, can we inject into that object via DyanmicMiddleware too?

**Reply by EskiMojo14** [reply]
if you save the reference, you can mutate it as needed

```ts
const thunkExtra = {} as { service?: Service }

const store = configureStore({
  reducer,
  middleware: gDM => gDM({ thunk: { extraArgument: thunkExtra } })
})

function initService() {
  thunkExtra.service = new Service()
}
```
